### ISSUE TITLE
Bug: Entity Reference field referencing unpublished node

### ISSUE THREAD

**Issue by lukewertz** [RESTful-Drupal/restful#683]
When an entityreference field has a cardinality of 1 and is referencing a node that is unpublished, the result from RESTful is that the entity simply disappears from the return. I would expect to get back a null value, not that the property would disappear.

**Comment by e0ipso** [comment]
I agree with that. It seems that `NULL` values are an edge case that is not very well handled.

**Comment by lukewertz** [comment]
Actually, I think this bug report might be a duplicate of https://github.com/RESTful-Drupal/restful/issues/492

If I'm able to this week, I'll try pulling and testing it... and perhaps adding a test for it as well.

**Comment by amitaibu** [comment]
Yeah, we should probably check not only isset,  but also array_key_exists - to better distinguish null from non existing
